portfolio/src/main/webapp/script.js

```diff
@@ -212,3 +219,23 @@ function deleteComments() {
     document.getElementById("commentsContainer").innerHTML = '';
   });
 }
+
+let userIsLoggedIn = false;
+
+/* Display the login logout display by fetching from server */
+function fetchLoginLogoutAndShowOrHideComments() {
+  fetch('/account').then(response => response.text()).then(html => {
+    document.getElementById("loginLogoutContainer").innerHTML = html;
+    userIsLoggedIn = html.includes("Logout ") ? true : false;
```

Comment:
Searching the response content is a fragile way to initialize local state; in a real-world application, someone might change the text in LoginLogoutServlet for stylistic reasons and unknowingly break this functionality.

A better approach might be to return a JSON response from your servlet, and handle presentation exclusively in Javascript. e.g. your servlet could simply return one of:

```
{"loginURL" : "..."}
// or
{"loggedInEmail": "...", "logoutURL" : "..."}
```

And in Javascript, you can initialize `userIsLoggedIn` based on whether `loggedInEmail` is populated, and also render the login/logout forms (which keeps HTML together with the other front-end code).

Reply:
Took me a while to figure out this one! Didn't notice that json properties had to have quotations too... changes made though!